mediator/med_methods_mod.F90

```diff
@@ -737,11 +738,16 @@ subroutine med_methods_FB_reset(FB, value, rc)
       lvalue = value
     endif
 
-    call ESMF_FieldBundleGet(FB, fieldCount=fieldCount, rc=rc)
-    if (chkerr(rc,__LINE__,u_FILE_u)) return
-    allocate(lfieldnamelist(fieldCount))
-    call ESMF_FieldBundleGet(FB, fieldNameList=lfieldnamelist, rc=rc)
-    if (chkerr(rc,__LINE__,u_FILE_u)) return
+    if (ESMF_FieldBundleIsCreated(fieldbundle=FB)) then
+       call ESMF_FieldBundleGet(FB, fieldCount=fieldCount, rc=rc)
+       if (chkerr(rc,__LINE__,u_FILE_u)) return
+       allocate(lfieldnamelist(fieldCount))
+       call ESMF_FieldBundleGet(FB, fieldNameList=lfieldnamelist, rc=rc)
+       if (chkerr(rc,__LINE__,u_FILE_u)) return
+    else
+       fieldCount=0
+       allocate(lfieldnamelist(fieldCount))
```

Comment:
I think that some compilers will throw an error on allocate with count=0

Reply:
Section 2.4.5 of the F90 standard says that arrays can have zero size so I hope that all compilers can handle allocate with count=0.
[F90 Standard](https://wg5-fortran.org/N001-N1100/N692.pdf)
Do you still want me to change this? It just means that I have to check before the deallocation and make sure that the array isn't used at all.

Reply:
I remember having problems with this but don't remember which compiler or how far back it was, you can leave this one.